packages/phone-number-privacy/combiner/src/utils/no-quota-cache.ts

```diff
@@ -0,0 +1,31 @@
+import { LRUCache } from 'lru-cache'
+
+// Stores if an address does not have more quota with its totalQuota number
+export class NoQuotaCache {
+  private cache: LRUCache<string, number, any>
+  constructor(seconds: number = 5) {
+    this.cache = new LRUCache({
+      max: 500,
+      ttl: seconds * 1000,
+      allowStale: false,
+    })
+  }
+
+  maximumQuouaReached = (address: string): boolean => {
+    const dek = this.cache.get(address)
+
+    return dek !== undefined
```

Comment:
Is this right? If there's no record in the cache how do we know if max quota was reached?

Reply:
ahh I see so if totalQuota is stored in the cache we know performedQueryCount === totalQuota

Reply:
exactly